Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/GUIGamePanel.java

```diff
@@ -0,0 +1,134 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+import javax.swing.JPanel;
+
+@SuppressWarnings("serial")
+public class GUIGamePanel extends JPanel {
+	private String pressedKey;
+	private int hangmanStage;
+
+	public GUIGamePanel(){
+		super();
+		KeyListener listener = new KeyListener() {
+			@Override
+			public void keyTyped(KeyEvent e) {
+			}
+
+			@Override
+			public void keyPressed(KeyEvent e) {
+				//System.out.println("keyPressed="+KeyEvent.getKeyText(e.getKeyCode()));
+				String key=KeyEvent.getKeyText(e.getKeyCode());
+				switch (key){
+				case "Shift":
+				case "Windows":
+				case "Ctrl":
+				case "Alt":
+				case "Caps Lock":
+				case "Space":
+				case "Comma":
+				case "0":
+				case "1":
+				case "2":
+				case "3":
+				case "4":
+				case "5":
+				case "6":
+				case "7":
+				case "9":
+				case "8":
+				case "Minus":
+				case "Equals":
+				case "Backspace":
+				case "Open Bracket":
+				case "Close Bracket":
+				case "Semicolon":
+				case "Quote":
+				case "Slash":
+				case "Back Slash":
+				case "Alt Graph":
+				case "Context Menu":
+
+					return;
+				default:
+					pressedKey= key;
+				}
+
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+			}
+		};
+		addKeyListener(listener);
+		setFocusable(true);
+	}
+
+	public String getPressedKey(){
```

Comment:
I think logically it makes more sense if the panel were to return the symbol from the last **legal** pressed key. In which case, the field pressedKey and the return type of getPressedKey() would logically be char, as that would guarantee you have a single character to input to the game. To deal with the case of there being no keys pressed yet, I would actually make the return value Optional<Char> (Optional libraries exist and I believe are built into Java 8) which would be cleaner than throwing an exception (which would still be an alternative)